ocaml/xapi/api_server.ml

```diff
@@ -196,11 +208,11 @@ let jsoncallback req bio _ =
   let fd = Buf_io.fd_of bio in (* fd only used for writing *)
   let body = Http_svr.read_body ~limit:Db_globs.http_limit_max_rpc_size req bio in
   try
-    let rpc = Jsonrpc.call_of_string body in
-    let response = Jsonrpc.a_of_response
+    let version, id, rpc = Jsonrpc.version_id_and_call_of_string body in
+    let response = Jsonrpc.a_of_response ~id ~version
         ~empty:Bigbuffer.make
         ~append:(fun buf s -> Bigbuffer.append_substring buf s 0 (String.length s))
-        (callback1 false req fd (Some body) rpc) in
+        (callback1 ~json_ver:version true req fd (Some body) rpc) in
```

Comment:
Then I need the label in line 212, but this is ok, better have the parameter of callback1 named properly.